### PR TITLE
Simple Payments: Improve error messages

### PR DESCRIPTION
**Pull request by iamtakashi**
The use of "can not" is grammatically incorrect in this context. It should be "cannot".

### CODE REVIEW

File: client/components/tinymce/plugins/simple-payments/dialog/form.jsx [hunk 1]
```diff
@@ -105,14 +105,14 @@ const validate = ( values, props ) => {
...
-			errors.price = translate( 'Price can not have more than %(countDecimal)s.', {
+			errors.price = translate( 'Price cannot have more than %(countDecimal)s.', {
 				args: { countDecimal },
 			} );
 		}
 	}
 
 	if ( ! values.email ) {
-		errors.email = translate( 'Email address can not be empty.' );
+		errors.email = translate( 'Email address cannot be empty.' );
```
Comment (a8ci18n) [inline client/components/tinymce/plugins/simple-payments/dialog/form.jsx]: :disappointed: 24 existing translations will be lost with this change.
If the string must change, the following option has already been translated **26** times:
`translate( 'Email address should not be empty' )` *ES Score: 7*

Help me improve these suggestions: react with :thumbsdown: if the suggestion doesn't make any sense, or with :thumbsup: if it's a particularly good one (even if not implemented).

File: client/components/tinymce/plugins/simple-payments/dialog/form.jsx [hunk 2]
```diff
@@ -105,14 +105,14 @@ const validate = ( values, props ) => {
 					},
 				}
 			);
-			errors.price = translate( 'Price can not have more than %(countDecimal)s.', {
+			errors.price = translate( 'Price cannot have more than %(countDecimal)s.', {
```
Comment (a8ci18n) [inline client/components/tinymce/plugins/simple-payments/dialog/form.jsx]: :ok: This change is safe, we'll keep the existing 19 translations.

File: client/components/tinymce/plugins/simple-payments/dialog/form.jsx [hunk 3]
```diff
@@ -82,18 +82,18 @@ const validate = ( values, props ) => {
...
+		errors.price = translate( 'Price cannot be empty.' );
 	} else if ( parseFloat( values.price ) === NaN ) {
 		errors.price = translate( 'Invalid price' );
 	} else if ( parseFloat( values.price ) < 0 ) {
-		errors.price = translate( 'Price can not be negative.' );
+		errors.price = translate( 'Price cannot be negative.' );
 	} else if ( decimalPlaces( values.price ) > precision ) {
 		if ( precision === 0 ) {
-			errors.price = translate( 'Price can not have decimal places.' );
+			errors.price = translate( 'Price cannot have decimal places.' );
```
Comment (a8ci18n) [inline client/components/tinymce/plugins/simple-payments/dialog/form.jsx]: :ok: This change is safe, we'll keep the existing 20 translations.

File: client/components/tinymce/plugins/simple-payments/dialog/form.jsx [hunk 4]
```diff
@@ -82,18 +82,18 @@ const validate = ( values, props ) => {
...
 	}
 
 	if ( ! values.price ) {
-		errors.price = translate( 'Price can not be empty.' );
+		errors.price = translate( 'Price cannot be empty.' );
 	} else if ( parseFloat( values.price ) === NaN ) {
 		errors.price = translate( 'Invalid price' );
 	} else if ( parseFloat( values.price ) < 0 ) {
-		errors.price = translate( 'Price can not be negative.' );
+		errors.price = translate( 'Price cannot be negative.' );
```
Comment (a8ci18n) [inline client/components/tinymce/plugins/simple-payments/dialog/form.jsx]: :ok: This change is safe, we'll keep the existing 20 translations.

File: client/components/tinymce/plugins/simple-payments/dialog/form.jsx [hunk 5]
```diff
@@ -82,18 +82,18 @@ const validate = ( values, props ) => {
 	const errors = {};
 
 	if ( ! values.title ) {
-		errors.title = translate( 'This can not be empty.' );
+		errors.title = translate( 'This cannot be empty.' );
 	}
 
 	if ( ! values.price ) {
-		errors.price = translate( 'Price can not be empty.' );
+		errors.price = translate( 'Price cannot be empty.' );
```
Comment (a8ci18n) [inline client/components/tinymce/plugins/simple-payments/dialog/form.jsx]: :ok: This change is safe, we'll keep the existing 18 translations.

File: client/components/tinymce/plugins/simple-payments/dialog/form.jsx [hunk 6]
```diff
@@ -82,18 +82,18 @@ const validate = ( values, props ) => {
 	const errors = {};
 
 	if ( ! values.title ) {
-		errors.title = translate( 'This can not be empty.' );
+		errors.title = translate( 'This cannot be empty.' );
```
Comment (a8ci18n) [inline client/components/tinymce/plugins/simple-payments/dialog/form.jsx]: Hi! I've found a possible matching string that has already been translated **70** times:
`translate( 'Passwords cannot be empty.' )` *ES Score: 8*
See 1 additional suggestion in the [PR translation status page](https://translate.wordpress.com/localci/status/automattic/wp-calypso/update/error-message-simple-payments)

Help me improve these suggestions: react with :thumbsdown: if the suggestion doesn't make any sense, or with :thumbsup: if it's a particularly good one (even if not implemented).

**Comment by lamosty** [comment]
Looks good. Have it been reviewed by somebody from editorial? Also, what about "can't"? I know we use informal language throughout Calypso so might be also a good fit.

**Comment by benhuberman** [comment]
Hey @iamtakashi  -- for some reason I can't see the error message when going to the testing link above. A screenshot could definitely help, though I can confirm that "can not" is incorrect, and depending on the context we could go with either **cannot** or **can't**. 

I might change my mind when I see this in context, but I'd probably also recommend adding a noun after "This" to make sure there's no ambiguity, e.g. "This field can't be empty," etc.

**Comment by iamtakashi** [comment]
@benhuberman My apologies for the lack of screenshots. The following should cover all the changes proposed.

screen shot 2017-11-24 at 12 05 37

screen shot 2017-11-24 at 12 05 04

screen shot 2017-11-24 at 12 04 49

> I might change my mind when I see this in context, but I'd probably also recommend adding a noun after "This" to make sure there's no ambiguity, e.g. "This field can't be empty," etc.

That's a good idea. Please let me know if you still think it's a good idea after seeing the screenshot above.

**Comment by benhuberman** [comment]
Thanks for the screenshot, @iamtakashi  -- that really helps. Now that I see these in context, I'd actually suggest a slightly different approach, since "X cannot be empty" feels pretty cold and technical. I'd also go for slightly different phrasings when possible to make it sound less monotonous (though this is a minor point, since I'm guessing that most people won't have _all_ error messages displayed at once).

Here are some ideas -- I've included both "straight-up" and "more fun" alternatives:

- For "What is this payment for?"
> Please fill out this field.

> We'll need this information to proceed.

More fun:
> People need to know what they're paying for! Please add a brief description. 

- For _Price_:
> We'll need this information to proceed.

> Please add a price.
More fun:
> Everything comes with a price tag these days. Add yours here. 

- For _Email_:
> Please add an email address.

> We'll need an email address to proceed.

More fun:
> We want to make sure payments reach you, so please add an email address.

- For decimals:
> Prices can't contain decimal quantities.

> Prices can't contain decimal values.

More fun:
> We know every penny counts, but prices can't contain decimal values.

- For negative prices:
> Prices can't contain negative values.

More fun:
> Your price is negative -- now that doesn't sound right, does it?

Let me know if any of those doesn't work and I'll be happy to take another shot.

**Comment by iamtakashi** [comment]
Thanks, @benhuberman! (Sorry for the late response. I was on support last week). I've tried those fun ones. Does it look good to go?

screen shot 2017-12-04 at 16 55 04

screen shot 2017-12-04 at 16 56 09

screen shot 2017-12-04 at 16 58 35

**Comment by benhuberman** [comment]
It looks good to me, @iamtakashi!

**Pull request merged**